server/src/collection.rs

```diff
@@ -75,11 +75,60 @@ impl Collection {
     #[zbus(out_args("item", "prompt"))]
     pub async fn create_item(
         &self,
-        _properties: Properties,
-        _secret: SecretInner,
-        _replace: bool,
-    ) -> Result<(OwnedObjectPath, ObjectPath), ServiceError> {
-        todo!()
+        properties: Properties,
+        secret: SecretInner,
+        replace: bool,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
+        #[zbus(signal_emitter)] signal_emitter: zbus::object_server::SignalEmitter<'_>,
+    ) -> Result<(OwnedObjectPath, OwnedObjectPath), ServiceError> {
+        let SecretInner(session, iv, secret, _content_type) = secret;
+        let label = properties.label();
+        let attributes = properties.attributes().unwrap();
+        let manager = self.manager.lock().await;
+
+        let Some(session) = manager.session(&session) else {
+            tracing::error!("The session `{}` does not exist.", session);
+            return Err(ServiceError::NoSession(format!(
+                "The session `{}` does not exist.",
+                session
+            )));
+        };
+
+        let secret = match session.aes_key() {
+            Some(key) => oo7::crypto::decrypt(secret, &key, &iv),
+            None => zeroize::Zeroizing::new(secret),
+        };
+
+        let item = match self
+            .keyring
+            .create_item(label, &attributes, secret, replace)
+            .await
+        {
+            Ok(item) => item,
+            Err(_) => todo!(), // TODO: may be new ServiceError
+        };
+
+        let n_items = *self.item_index.read().await;
+        let item = item::Item::new(
+            item,
+            self.is_locked().await,
```

Comment:
Can you create an item on a locked collection? i don't think so, so it is safe to assume this would be unlocked. Leave a comment explaining why and add the usual check if the collection.is_locked & error message

Reply:
Yeah. I think passing `false` here is enough. Because, when calling CreateItem, a call to Service.Unlock happens before, automatically. After the Service.Unlock call returns CreateItem is on.

Reply:
>  Because, when calling CreateItem, a call to Service.Unlock happens before, automatically. After the Service.Unlock call returns CreateItem is on.

What do you mean? that is not true at all. The user is expected to unlock the collection, it is not done automatically

Reply:
I meant when calling CreateItem, regardless of the current locked status, a call to Service.Unlock happens beforehand (I'm not calling this so, automatically). When that call returns CreateItem starts. This is what I see with Bustle (with gnome-keyring-daemon and now  and with oo7-daemon)  :)

Reply:
I don't get what you mean still. Who calls unlock?